### PR TITLE
CLOUDP-64616: Replace deprecated cloud backup with new cloud backup

### PR DESCRIPTION
**Pull request by gssbzn**
## Proposed changes

_Jira ticket:_ [CLOUDP-64616](https://jira.mongodb.org/browse/CLOUDP-64616)

## Checklist

- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run `make fmt` and formatted my code

## Further comments

```bash
mongocli atlas backup restore list myDemo
{
	"links": [
		{
			"rel": "self",
			"href": "https://cloud-dev.mongodb.com/api/atlas/v1.0/groups/x/clusters/myDemo3/backup/restoreJobs?pageNum=1\u0026itemsPerPage=100"
		}
	],
	"results": [
		{
			"id": "x",
			"snapshotId": "x",
			"deliveryType": "download",
			"expired": true,
			"expiresAt": "2020-06-02T23:19:27Z",
			"finishedAt": "2020-06-02T23:19:31Z",
			"links": [
				{
					"rel": "self",
					"href": "https://cloud.mongodb.com/api/atlas/v1.0/groups/x/clusters/myDemo3/backup/restoreJobs/x"
				},
				{
					"rel": "http://mms.mongodb.com/snapshot",
					"href": "https://cloud.mongodb.com/api/atlas/v1.0/groups/x/clusters/myDemo3/backup/snapshots/x"
				},
				{
					"rel": "http://mms.mongodb.com/cluster",
					"href": "https://cloud.mongodb.com/api/atlas/v1.0/groups/x/clusters/myDemo3"
				},
				{
					"rel": "http://mms.mongodb.com/group",
					"href": "https://cloud.mongodb.com/api/atlas/v1.0/groups/x"
				}
			],
			"timestamp": "2020-06-02T19:00:49Z"
		}
	],
	"totalCount": 1
}
```

```bash
ongocli atlas backup restore start download --clusterName myDemo3 --snapshotId x 
{
	"id": "x",
	"snapshotId": "x",
	"deliveryType": "download",
	"expiresAt": "2020-06-03T13:34:59Z",
	"links": [
		{
			"rel": "self",
			"href": "https://cloud.mongodb.com/api/atlas/v1.0/groups/5e429f2e06822c6eac4d59c9/clusters/myDemo3/backup/restoreJobs/5ed76ec3b1d504005ed3305e"
		},
		{
			"rel": "http://mms.mongodb.com/snapshot",
			"href": "https://cloud.mongodb.com/api/atlas/v1.0/groups/x/clusters/myDemo3/backup/snapshots/x"
		},
		{
			"rel": "http://mms.mongodb.com/cluster",
			"href": "https://cloud.mongodb.com/api/atlas/v1.0/groups/x/clusters/myDemo3"
		},
		{
			"rel": "http://mms.mongodb.com/group",
			"href": "https://cloud.mongodb.com/api/atlas/v1.0/groups/x"
		}
	],
	"timestamp": "2020-06-02T19:00:49Z"
}
```

```bash
mongocli atlas backup snapshots ls myDemo3 -P dev
{
	"results": [
		{
			"id": "5ed7295c886631739d2c5010",
			"createdAt": "2020-06-03T04:39:59Z",
			"expiresAt": "2020-06-04T04:40:14Z",
			"links": [
				{
					"rel": "self",
					"href": "https://cloud.mongodb.com/api/atlas/v1.0/groups/x/clusters/myDemo3/backup/snapshots/x"
				},
				{
					"rel": "http://mms.mongodb.com/cluster",
					"href": "https://cloud.mongodb.com/api/atlas/v1.0/groups/x/clusters/myDemo3"
				}
			],
			"mongodVersion": "4.0.18",
			"snapshotType": "scheduled",
			"status": "completed",
			"storageSizeBytes": 4493111296,
			"type": "replicaSet"
		},
		{
			"id": "x",
			"createdAt": "2020-06-02T19:00:49Z",
			"expiresAt": "2020-06-05T19:01:01Z",
			"description": "dwa",
			"links": [
				{
					"rel": "self",
					"href": "https://cloud.mongodb.com/api/atlas/v1.0/groups/5e429f2e06822c6eac4d59c9/clusters/myDemo3/backup/snapshots/5ed6a174f78e45397e4a313d"
				},
				{
					"rel": "http://mms.mongodb.com/cluster",
					"href": "https://cloud.mongodb.com/api/atlas/v1.0/groups/x/clusters/myDemo3"
				}
			],
			"mongodVersion": "4.0.18",
			"snapshotType": "onDemand",
			"status": "completed",
			"storageSizeBytes": 4473647104,
			"type": "replicaSet"
		}
	],
	"links": [
		{
			"rel": "self",
			"href": "https://cloud.mongodb.com/api/atlas/v1.0/groups/x/clusters/myDemo3/backup/snapshots?pageNum=1\u0026itemsPerPage=100"
		}
	],
	"totalCount": 2
}
```

### CODE REVIEW

File: internal/store/continuous_backup.go [hunk 1]
```diff
@@ -37,14 +37,10 @@ type ContinuousJobCreator interface {
 	CreateContinuousRestoreJob(string, string, *atlas.ContinuousJobRequest) (*atlas.ContinuousJobs, error)
 }
 
-type SnapshotsLister interface {
+type ContinuousSnapshotsLister interface {
 	ContinuousSnapshots(string, string, *atlas.ListOptions) (*atlas.ContinuousSnapshots, error)
 }
 
-type SnapshotDescriber interface {
```
Comment (andreaangiolillo) [inline internal/store/continuous_backup.go:44]: [q] I am wondering why we are removing this. Do we have another API call for describing a snapshot?

Comment (gssbzn) [reply internal/store/continuous_backup.go:44]: short answer, we don't have a command using it, if we need it we'll add it later

**Comment by andreaangiolillo** [comment]
The task [cloud_manager_e2e](https://evergreen.mongodb.com/task/mongocli_master_e2e_cloud_manager_e2e_patch_54746059f08f2864fb524735d84ce2c5c6ae79ee_5ed76ea661837d082a5c41ed_20_06_03_09_34_31) is failing because this link [ca_certificates](http://us-east-1.ec2.archive.ubuntu.com/ubuntu/pool/main/c/ca-certificates/ca-certificates_20170717~16.04.2_all.deb) does not work anymore

**Comment by gssbzn** [comment]
@andreaangiolillo looks like an ubuntu issue, going to merge and wait

**Pull request merged**